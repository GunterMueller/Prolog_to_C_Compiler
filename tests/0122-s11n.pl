:- include(library(s11n)).
:- include(library(canon)).

main :-
	serialize_term(this(is,"a",[123]), X),
%	atom_codes(X, LST),
%	forall(member(Y, LST), put(Y)),
%	nl,
	deserialize_term(X, Z),
	writeq(Z), nl,
	Q = f(Q),
	serialize_term(Q, QS),
%	atom_codes(QS, QLST),
%	forall(member(QY, QLST), put(QY)),
%	nl,
	deserialize_term(QS, QZ),
	\+acyclic_term(QZ),
	canonical_term(QZ, _, P),
	writeq(P), nl.
