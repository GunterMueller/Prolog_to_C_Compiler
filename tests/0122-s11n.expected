this(is,[97],[123])
f(cycle_at_depth(0))
